esmvaltool/cmorizers/data/downloaders/wget.py

```diff
@@ -0,0 +1,106 @@
+"""wget based downloader."""
+
+import logging
+import os
+import subprocess
+
+from .downloader import BaseDownloader
+
+logger = logging.getLogger(__name__)
+
+
+class WGetDownloader(BaseDownloader):
+    """Data downloader based on wget."""
+    def download_folder(self, server_path, wget_options):
+        """Download folder.
+
+        Parameters
+        ----------
+        server_path: str
+            Path to remote folder
+        wget_options: list(str)
+            Extra options for wget
+        """
+        if self.overwrite:
+            raise ValueError(
+                'Overwrite does not work with downloading directories through '
+                'wget. Please, remove the unwanted data manually')
+        command = ['wget'] + wget_options + self.overwrite_options + [
+            f'--directory-prefix={self.local_folder}',
+            '--recursive',
+            '--no-directories',
+            f'{server_path}',
+        ]
+        logger.debug(command)
+        subprocess.check_output(command)
+
+    def download_file(self, server_path, wget_options):
+        """Download file.
+
+        Parameters
+        ----------
+        server_path: str
+            Path to remote file
+        wget_options: list(str)
+            Extra options for wget
+        """
+
+        command = ['wget'] + wget_options + self.overwrite_options + [
+            f'--directory-prefix={self.local_folder}',
+            '--no-directories',
+            server_path,
+        ]
+        if self.overwrite:
+            command.append(f'-O {os.path.basename(server_path)}')
+        logger.debug(command)
+        subprocess.check_output(command)
+
+    @property
+    def overwrite_options(self):
+        """Get overwrite options as configured in downloader."""
+        if not self.overwrite:
+            return [
+                '--no-clobber',
+            ]
+        return []
+
+
+class NASADownloader(WGetDownloader):
+    """Downloader for the NASA repository."""
+    def __init__(self, config, dataset, overwrite):
+        super().__init__(config, dataset, overwrite)
+        self.tier = 3
+
+        self._wget_common_options = [
+            "--load-cookies=~/.urs_cookies",
+            "--save-cookies=~/.urs_cookies",
+            "--auth-no-challenge=on",
+            "--keep-session-cookies",
+        ]
+
```

Comment:
I have tried all 4 downloader scripts based on this class:

- AURA-TES: network down, impossible to download at the moment
- MERRA2: only works when using the `urs.earthdata.nasa.gov` credentials in a `.netrc` file which is good since this is Tier3 data
- NSIDC-0116-*h: can be downloaded without a netrc file. I'm not sure this is something to worry. In fact `wget --no-check-certificate filename.nc` gets you the files which is not true for MERRA2.

Reply:
Do you remove the entries on the .netrc for the NSIDC data? I think this is also downloaded from NASA, so it may work with the ones you provided for the MERRA2

Reply:
In fact I managed to download the NSIDC data after removing the `.netrc` from my home directory on Mistral but not the MERRA2 data. It seems the requirements to get MERRA2 data are stricter than for NSIDC. I could download NSIDC data on my laptop (no `.netrc`) with `wget --no-check-certificate` but no MERRA2 data. I think NSIDC data are "less Tier3" than MERRA2. Nevertheless, it could be good to identify the wget option that enables downloading NSIDC to easily and fix that.

Reply:
I will take a look at it and try to restrict the NSIDC download, but it looks more like an error at their side